numpy/core/src/multiarray/arrayobject.c

```diff
@@ -661,36 +534,42 @@ static PyObject *
 array_repr(PyArrayObject *self)
 {
     PyObject *s, *arglist;
+    PyObject *reprfunc = NULL;
 
-    if (PyArray_ReprFunction == NULL) {
-        s = array_repr_builtin(self, 1);
+    npy_cache_import("numpy.core._internal", "_pyarray_repr_function", &reprfunc);
```

Comment:
Thanks for the explanations. My guess is this is not noticeable speedwise and not so much code, so +1 (module maybe a pendatic check that the code can really be deleted, i.e. no python func being set should never happen)-